snuba/subscriptions/subscription.py

```diff
@@ -0,0 +1,65 @@
+from datetime import datetime, timedelta
+from typing import Sequence, Tuple
+from uuid import uuid1
+
+from snuba.datasets.dataset import Dataset
+from snuba.query.query import Aggregation
+from snuba.query.types import Condition
+from snuba.redis import redis_client
+from snuba.subscriptions.data import Subscription
+from snuba.subscriptions.partitioner import DatasetSubscriptionPartitioner
+from snuba.subscriptions.store import RedisSubscriptionStore
+from snuba.utils.metrics.timer import Timer
+from snuba.web.query import parse_and_run_query
+
+
+class InvalidSubscriptionError(Exception):
+    pass
+
+
+class SubscriptionCreator:
+    """
+    Handles creation of a `Subscription`, including assigning an ID and validating that
+    the resulting query is valid.
+    """
+
+    def __init__(self, dataset: Dataset):
+        self.dataset = dataset
+
+    def create(
+        self,
+        project_id: int,
+        conditions: Sequence[Condition],
+        aggregations: Sequence[Aggregation],
+        time_window: timedelta,
+        resolution: timedelta,
```

Comment:
That's a good justification for removing it... As long as we're able to still pass the subscription id to the results producer then we're fine, since I need that id to match updates to subscriptions on my end. We'll need to change the subscription store to return the subscription id and subscription when fetching all of them, which isn't a huge deal